### ISSUE TITLE
vscode.executeCodeActionProvider returns no items in latest Insiders build

### ISSUE THREAD

**Issue by DanTup** [microsoft/vscode#199548]
The following code registers a Code Actions provider that returns a single code action then executes it using `vscode.executeCodeActionProvider`. It prints the number of code actions returned from the provider and the number received from the command.

```ts
import * as vscode from 'vscode';

export async function activate(context: vscode.ExtensionContext) {
	const filePath = "C:\\Dev\\Dart-Code\\Dart-Code\\src\\test\\test_projects\\hello_world\\lib\\empty.dart";
	context.subscriptions.push(vscode.languages.registerCodeActionsProvider(
		{ scheme: 'file' },
		{
			provideCodeActions(document, range, context, token) {
				const results = [
					new vscode.CodeAction("Foo", vscode.CodeActionKind.QuickFix),
				];

				console.log(`Returning ${results.length} CodeActions`);
				return results;
			},
		}
	));
	await null;
	const doc = await vscode.workspace.openTextDocument(filePath);
	const editor = await vscode.window.showTextDocument(doc);
	const codeActions = await vscode.commands.executeCommand<vscode.CodeAction[]>("vscode.executeCodeActionProvider", editor.document.uri, new vscode.Range(doc.positionAt(0), doc.positionAt(0)));
	console.log(`Got ${codeActions.length} CodeActions`);

}

export function deactivate() { }
```

The output with current stable VS Code is:

```
Returning 1 CodeActions
Got 1 CodeActions
```

However in Insiders, the output is:

```
Returning 1 CodeActions
Got 0 CodeActions
```

**Comment by justschen** [comment]
From my own testing in a custom extension, I wasn't able to reproduce this. I don't think there were any major changes that modified the `CodeActionSet` or `ManagedCodeActionSet`.

![Screenshot 2023-12-04 at 4 10 51 PM](https://github.com/microsoft/vscode/assets/54879025/c84e3022-2832-449c-bb2b-36ca59df958b)
<img width="948" alt="Screenshot 2023-12-04 at 4 16 24 PM" src="https://github.com/microsoft/vscode/assets/54879025/4f580ce5-bca0-4a93-9034-e7387761838c">

Not sure if there are any other additional details besides the tests that I see from the above commit

**Comment by DanTup** [comment]
hmmm, there's something odd going on here. I changed my test code to use `untitled` instead of files, and it worked fine:

![image](https://github.com/microsoft/vscode/assets/1078012/e9c7f697-50cc-4de8-a2ca-d6bd77adc36b)

So I changed it back to file and used an `md` file, and it also worked fine.

So I changed it back to a Dart file, and it's broken again:

![image](https://github.com/microsoft/vscode/assets/1078012/9098e044-87ff-4660-b984-eb393513c71d)

I disabled the Dart extension to ensure that wasn't interfering with it (as shown in the screenshot above), but I still see it fail.

Can you try with a `.dart` file to see if it repros for you? I don't see why that should make a difference, but it seems to for me and it's not clear why.

**Comment by DanTup** [comment]
Ok, I figured out the difference - it only happens the opened file is empty. If the file contains any characters, it works fine. If the document is empty, it fails.

Here's a self-contained repro using a text file:

https://github.com/DanTup/temp-vscode-repro-199548

**Comment by justschen** [comment]
currently still git bisecting, but I have a hunch it may be related to the latest change regarding code actions https://github.com/microsoft/vscode/pull/198864#event-11090390532 cc. @aiday-mar (maybe not this exact PR, but i am seeing between good and bad commits the lack of lightbulb/code action being detected)

**Comment by justschen** [comment]
TLDR on the issue:
Contributed code actions (and also code actions in general) are not shown in empty lines or zero length selections.  

Missing additional code actions and the action bar. In examples below, `Foo` is the contributed code action from a custom extension.

current:
![Screenshot 2023-12-05 at 5 02 37 PM](https://github.com/microsoft/vscode/assets/54879025/fb64f7b1-1106-4f63-a329-964d79cc8e21)
![Screenshot 2023-12-05 at 5 05 29 PM](https://github.com/microsoft/vscode/assets/54879025/1f8e8501-53d2-44b8-8400-163bec9ba0ad)

expected (from stable):
<img width="391" alt="Screenshot 2023-12-05 at 5 01 29 PM" src="https://github.com/microsoft/vscode/assets/54879025/15bacac4-466c-40d5-9425-18929146271b">
<img width="405" alt="Screenshot 2023-12-05 at 5 05 07 PM" src="https://github.com/microsoft/vscode/assets/54879025/6ccb7689-4268-47c6-9c92-5a905f44e537">

example from oss where we show non-fix actions in empty lines:
![Screenshot 2023-12-05 at 5 09 48 PM](https://github.com/microsoft/vscode/assets/54879025/9cd60841-f0c3-42f1-a8a4-cab801b20aa4)

**Comment by DanTup** [comment]
I'm curious what the reason was for hiding these. If a language doesn't want code actions on empty lines/selections, it could just not return them. Having VS Code doing additional filtering (without these being clearly specified) seems a little weird even if it's not clear how useful they are.

**Comment by justschen** [comment]
The main reason was regarding the new AI lightbulb and the changes that were made to have the lightbulb show on different scenarios (on code, always, or never). The default setting is that we don't want to show the lightbulb on empty lines, so there was some additional code that removed actions (including quick fixes or additional code actions) on empty lines or empty selections, which we just reverted for the latest release
